### ISSUE TITLE
Rename `libparsec::ClientConfig` to `PartialClientConfig` or `RawClientConfig`

### ISSUE THREAD

**Issue by FirelightFlagboy** [Scille/parsec-cloud#7897]
A note for later: could we rename this to something like `PartialClientConfig` or `RawClientConfig` to make things more explicit ?

_Originally posted by @AureliaDolo in https://github.com/Scille/parsec-cloud/pull/7896#discussion_r1701933651_

**Comment by touilleMan** [comment]
I agree with the need for a renaming (having multiple things with the same name is never a good idea)

I would go with `RawClientConfig` (given the configuration is not partial from the external user point of view)
